rpc/src/transaction.rs

```
@@ -280,6 +454,14 @@ impl WarningTxBuilder {
     pub fn seller_input_sighash(&self) -> Result<TapSighash> {
         self.key_spend_sighash(self.unsigned_tx()?, 1)
     }
+
+    pub fn compute_signed_tx(&mut self) -> Result<&mut Self> {
+        let tx = self.unsigned_tx()?.clone()
+            .with_key_spend_witness(0, self.buyer_input_signature()?)
+            .with_key_spend_witness(1, self.seller_input_signature()?);
+        self.signed_tx.get_or_insert(tx);
```

Is that right, that you are preparing here for not changing the existing signed_tx?

---

I've been using `get_or_insert` for all the tx-builder setters to try to improve the safety of the low level protocol API, by preventing any fields of the trade model from mutating (and potentially becoming inconsistent with each other) once they've already been initialised, say due to calling the same RPC method twice.

I think it may be good to go one step further than that and make the setters return an error if the variable gets set twice with different values, instead of silently ignoring the second value, but since that is a fairly extensive change, I didn't really want to include it in this PR.